wordpress-mobile/WordPress-Android
# [Dependency Updates] Remove Unused Dependencies for WordPress Module

Parent #17558
Partially Closes: #17555

This issue is about removing all unused dependencies for the the `WordPress` related module.

I recommend during a commit-by-commit review. You will also find additional information on the individual changes by reading a commit's description.

-----

Compose Dependencies List:

1. [Group androidx compose ui tooling dependencies together.](https://github.com/wordpress-mobile/WordPress-Android/commit/1da91a1e661be35838777a7448af26d32c7a9752)
2. [Split androidx lifecycle version and lifecycle compose version.](https://github.com/wordpress-mobile/WordPress-Android/commit/34084e864fcddd955690c640c393a8999846cd40)
3. [Split androidx compose version and compose compiler version.](https://github.com/wordpress-mobile/WordPress-Android/commit/376025532de0e1ff3c9baf108518d990378f4c06)
4. [Add compose related missing transitively used dependencies.](https://github.com/wordpress-mobile/WordPress-Android/commit/85d3f05604152a33574e51c1115fd12e9701823c)
5. [Regroup compose related dependencies.](https://github.com/wordpress-mobile/WordPress-Android/commit/6e07b0716980a3698599e5bf3728922f91ff01d2)

Libs Dependencies List:

1. [Remove unused email checker dependency.](https://github.com/wordpress-mobile/WordPress-Android/commit/bf87d34ee3c7d0542144ac93ac581c9f47f75b44)
2. [Remove unused graph view dependency.](https://github.com/wordpress-mobile/WordPress-Android/commit/b236b0c1ca18e88ac48d68abf429080e340ec925)

Main Dependencies List:

1. [Remove unused androidx legacy support v4 and core ui dependencies.](https://github.com/wordpress-mobile/WordPress-Android/commit/07c3837f343d8827f8fcdb4917e5e1b46e2a52de)
3. [Remove unused androidx exif interface dependency.](https://github.com/wordpress-mobile/WordPress-Android/commit/aff77589b9febf50de84215f85a316383fcf83a3)

Test Dependencies List:

1. [Remove unused androidx test core dependency.](https://github.com/wordpress-mobile/WordPress-Android/commit/434971eec0d4089c34d5e8c69288d538dbd3eac3)
2. [Remove unused bouncycastle dependency.](https://github.com/wordpress-mobile/WordPress-Android/commit/179121580112ad75715df3c1bfe686919cde2cdb)

Refactor List:

1. [Add associated stetho related comment for future reference.](https://github.com/wordpress-mobile/WordPress-Android/commit/4895c28bd90c10eb94f3464ffb53170e8f4889cd)
2. [Add eventbus related missing transitively used dependencies.](https://github.com/wordpress-mobile/WordPress-Android/commit/f0e039c7f27de67cd792b1c595435cca4703b9b2)

-----

@oguzkocer I added you alone as the reviewer of this change as this will help us progress with the dependency updates, in terms of making the list of dependencies that we need to update more straightforward for us. The other such PR was the #17605.

-----

To test:

1. There is nothing much to test here.
2. Verifying that all the CI checks are successful should be enough.
3. However, if you want to be thorough about reviewing this change, you could:
    1. See the dependency tree diff result and verify correctness.
    2. Quickly smoke test both, the WordPress and Jetpack apps, and see if they both work as expected.

## Regression Notes
1. Potential unintended areas of impact

Some of the dependencies removed might still be required as they might be hidden transitive dependencies and thus either crashing the app or causing some kind of misbehaviour.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section.

3. What automated tests I added (or what prevented me from doing so)

N/A

-----

PR submission checklist:
